#3576 beam_stuck_guard_improvement

edwazhao opened the pull request:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/lbnl-ucb-sti/beam/3630)
<!-- Reviewable:end -->

Adding two parameters to gradle settings:
min_cpu_usage, the min cpu usage when checking the stuck guard shell
max_inactive_time_interval, the max time interval for no logs output when checking the stuck guard shell

HoneyTauOverTwo committed:
Fixes issue #3457 of duplicate column names in passengerPerTripSubway.csv when the data contains only zeroes.

HoneyTauOverTwo committed:
Fixes issue #3457 of duplicate column names in passengerPerTripSubway.csv when the data contains only zeroes.

HoneyTauOverTwo committed:
Fixes indentation of bracket.

HoneyTauOverTwo committed:
Fixes back indentation of bracket. (having some trouble with git on IntelliJ)

alex-vv committed:
Drop unused variable

HoneyTauOverTwo committed:
Modifies the dataSet array dimensions to omit columns that would be filled with zeroes.

HoneyTauOverTwo committed:
Removes columns filled with zeroes from passengerPerTripCar.

HoneyTauOverTwo committed:
Removes all zero-filled columns

Applies a similar strategy which was made for CarPassengerPerTrip, previously, this only removed the last X zero-filled columns.

HoneyTauOverTwo committed:
Removes all zero-filled columns from the csv output passengerPerTripRideHail.

HoneyTauOverTwo committed:
Prevents possible bug if for some reason "process" gets called more than once by clearing "nonZeroColumns" list.

HoneyTauOverTwo committed:
Adds unit tests

to analyse iterations output csv files

alex-vv committed:
Memory optimizations

HoneyTauOverTwo committed:
Merging develop

HoneyTauOverTwo committed:
Merging develop

HoneyTauOverTwo committed:
Merging develop

HoneyTauOverTwo committed:
Merge branch 'develop' into 3457-output-duplicate-columns

HoneyTauOverTwo committed:
Merging develop

HoneyTauOverTwo committed:
Merging develop

HoneyTauOverTwo committed:
Merging develop

HoneyTauOverTwo committed:
Merging develop

HoneyTauOverTwo committed:
Merging develop

HoneyTauOverTwo committed:
Improves output files testing

alex-vv committed:
Fix tests

alex-vv committed:
Fix test

HoneyTauOverTwo committed:
Separates edge-cases testing

HoneyTauOverTwo committed:
Removes unnecessary extensions and unused function def

edwardTSC committed:
#3576 beam_stuck_guard_improvement

edwardTSC committed:
#3576 beam_stuck_guard_improvement

HoneyTauOverTwo committed:
Commit after running Beam to pass CI test

HoneyTauOverTwo committed:
Commit after running gradle fmt

nikolayilyin commented on aws/build.gradle:
```diff
@@ -112,6 +112,8 @@ task deploy(type: AWSLambdaInvokeTask) {
   "max_cores": ${ext.getParameterValue('maxCores') ?: 0},
   "min_memory": ${ext.getParameterValue('minMemory') ?: 0},
   "max_memory": ${ext.getParameterValue('maxMemory') ?: 0},
+  "min_cpu_usage": ${ext.getParameterValue('minCpuUsage') ?: 1},
+  "max_inactive_time_interval": ${ext.getParameterValue('maxInactiveTimeInterval') ?: 5},
```
I would change the names of parameters to include 'stuckGuard', i.e. stuckGuardMinCpuUsage and stuckGuardMaxInactiveTimeInterval to reduce the possible confusion.

edwazhao replied:
Hi @nikolayilyin , I have changed the parameter names, please take a look again.

For the documentation, I couldn't find the correct place, do you mean here?
https://beam.readthedocs.io/en/latest/developers.html#automated-cloud-deployment with the parameter part?

nikolayilyin replied:
Yes, the link you shared contains the description of 'deploy' command, let's add the description of new parameters to that comand

edwazhao replied:
Hi @nikolayilyin , I did a commit to add description of the deploy command.
And also I did a rebase from develop branch, but now the changes of PR looks mess, only three files is changed by me
![image](https://user-images.githubusercontent.com/11383105/194872583-2e1d8b66-a083-460b-ae4c-81ef6191abf1.png)

Do you have any suggestions of the default values? stuckGuardMinCpuUsage:1, stuckGuardMaxInactiveTimeInterval:5

nikolayilyin replied:
The default values looks OK for me, let's keep them as 5 and 1 for now

edwazhao replied:
Can we close the issue and merge the PR? Or what is the process for BEAM project?

nikolayilyin reviewed (changes requested):
Apart from a nitpick the changes also require to change the documentation, the deploy command is described in there - docs/developers.rst, let's add these two parameters into documentation

edwardTSC committed:
#3576 fix review with new parameter names

edwardTSC committed:
#3576 fix review with new parameter names

alex-vv committed:
Merge branch 'develop' into alex-vv/3609-memory-optimization

alex-vv committed:
Fix test

alex-vv committed:
Tag some tests as retryable

alex-vv committed:
Fix test

nikolayilyin committed:
copy cloud-init if beam was unable to start on AWS s3

HoneyTauOverTwo committed:
Adds retryable tag to two flaky tests.

nikolayilyin committed:
Merge pull request #3629 from LBNL-UCB-STI/alex-vv/3609-memory-optimization

Memory usage optimization

nikolayilyin committed:
Merge pull request #3635 from LBNL-UCB-STI/inm/save-cloud-init-when-failed-run-beam

Copy cloud-init if BEAM was unable to start on AWS s3

HoneyTauOverTwo committed:
Removes testing for exact header for beam simulations.

Turns out that testing the header leads to a flaky test.

Exact header is still tested in the edge-cases where we have full control of the output file.

It still checks if the columns were properly omitted which is the main point.

nikolayilyin committed:
Merge pull request #3620 from LBNL-UCB-STI/3457-output-duplicate-columns

Fixes issue #3457 of duplicate column names in passengerPerTripSubway.csv when the data contains only zeroes.

nikolayilyin committed:
a config key to change the timeout

Xuan-1998 committed:
Merge pull request #3638 from LBNL-UCB-STI/inm/urbansim-scenario-loading-timeout

Configurable timeout for urbansim scenario loading

edwardTSC committed:
#3576 fix review add more description to the deployment command

edwardTSC committed:
Merge remote-tracking branch 'origin/edward/#3576_beam_stuck_guard_improvement' into edward/#3576_beam_stuck_guard_improvement
